### PR TITLE
Practice4 

### PR DESCRIPTION
**Pull request by nick-krasnitskiy**
1. Add third-party libraries (ImageScrollView, Alamofire) using Cocoa Pods.
2. Create UI for 3 screens using Compositional Layout.
3. Add data saving to the Core Data model and transfer it to UI.

### CODE REVIEW

File: DogsList/.swiftlint.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ disabled_rules: # rule identifiers turned on by default to exclude from running
   - trailing_whitespace
   - identifier_name
   - type_name
+  - file_length
```
Comment (vasilevkin) [inline DogsList/.swiftlint.yml]: do not disable rules, use default settings and fix you code

File: DogsList/DogsList.xcodeproj/xcuserdata/nvkrasnitskiy.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 2]
```diff
@@ -7,7 +7,7 @@
 		<key>DogsList.xcscheme_^#shared#^_</key>
```
Comment (vasilevkin) [inline DogsList/DogsList.xcodeproj/xcuserdata/nvkrasnitskiy.xcuserdatad/xcschemes/xcschememanagement.plist:7]: do not include this file into pr

File: DogsList/DogsList.xcworkspace/contents.xcworkspacedata [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vasilevkin) [inline DogsList/DogsList.xcworkspace/contents.xcworkspacedata:1]: do not include this file into pr

File: DogsList/DogsList.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vasilevkin) [inline DogsList/DogsList.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: do not include this file into pr

**Review by vasilevkin — CHANGES_REQUESTED** [review]
do not include Pods into pr.
check/update gitignore

File: DogsList/DogsList.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 4 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Fixed.

File: DogsList/DogsList.xcworkspace/contents.xcworkspacedata (hunk 3 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList.xcworkspace/contents.xcworkspacedata:1]: Fixed.

File: DogsList/DogsList.xcodeproj/xcuserdata/nvkrasnitskiy.xcuserdatad/xcschemes/xcschememanagement.plist (hunk 2 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList.xcodeproj/xcuserdata/nvkrasnitskiy.xcuserdatad/xcschemes/xcschememanagement.plist:7]: Fixed.

File: DogsList/.swiftlint.yml (hunk 1 above)
Comment (nick-krasnitskiy) [reply DogsList/.swiftlint.yml]: Fixed.

**Review by vasilevkin — CHANGES_REQUESTED** [review]
this pr should contain changes for Practice 4 only.
also fix all merge conflicts

**Comment by nick-krasnitskiy** [comment]
> this pr should contain changes for Practice 4 only.
> also fix all merge conflicts

It contains only Practice 4 changes. Merge conflicts were fixed with Alexey Korol's help.

**Comment by vasilevkin** [comment]
> It contains only Practice 4 changes. Merge conflicts were fixed with Alexey Korol's help.

Great! You have a progress in studying.
But this pr still contains a pods/ code.
It should contain your code only, but not files from pods dependencies.

**Comment by nick-krasnitskiy** [comment]
> > It contains only Practice 4 changes. Merge conflicts were fixed with Alexey Korol's help.
> 
> Great! You have a progress in studying.
> But this pr still contains a pods/ code.
> It should contain your code only, but not files from pods dependencies.

Fixed.

**Comment by vasilevkin** [comment]
it's much better now.
continue to implement the project,
you're on the correct way!

**Comment by nick-krasnitskiy** [comment]
> it's much better now.
> continue to implement the project,
> you're on the correct way!

Thaks a lot. Should I do something else to approve this PR?

**Comment by vasilevkin** [comment]
> > it's much better now.
> > continue to implement the project,
> > you're on the correct way!
> 
> Thaks a lot. Should I do something else to approve this PR?

It's called 
Practice4 (not finished yet)

at least update this title when it's finally ready

**Comment by nick-krasnitskiy** [comment]
> > > it's much better now.
> > > continue to implement the project,
> > > you're on the correct way!
> > 
> > 
> > Thaks a lot. Should I do something else to approve this PR?
> 
> It's called
> Practice4 (not finished yet)
> 
> at least update this title when it's finally ready

Thanks. Fixed.

**Review by vasilevkin — APPROVED** [review]
LGTM

**Pull request merged**